kyo-core/shared/src/main/scala/kyo/KyoApp.scala

```diff
@@ -43,8 +41,18 @@ object KyoApp:
             this._args = args
             for proc <- initCode do proc()
 
+        protected def exit(code: Int): Unit = kernel.Platform.exit(code)
+
+        protected def onResult(result: Result[Any, Any]): Unit =
+            if !result.exists(().equals(_)) then
+                println(pprint.apply(result).plainText)
+            result match
+                case Result.Error(_) => exit(1)
+                case _               => ()
```

Comment:
```suggestion
                case _               => exit(0)
```
Maybe this could be like this to make it super regular and predictable?

Reply:
A `KyoApp` can contain multiple runs. We should only exit on failure to prevent prematurely exiting.

Reply:
> can contain multiple runs

Oh, that caught me by surprise, I didn't know that could be the case.

On another note, it's it preferable to throw an exception in a situation like this? Such exception shouldn't be handled in the program and should result in its termination. I assume that all our target runtime (JVM, Scala Native and Scala.js) will return a correct, i.e. non-zero, code on terminating like this.

Reply:
The exception will already be printed via the pretty print. We could consider printing only success and re-throwing exceptions.

The issue is `Result[Any, A]` may contain other `Fail` values than `Throwable`.

Reply:
> The issue is Result[Any, A] may contain other Fail values than Throwable.

It's possible to pattern match to handle exceptions differently:
```scala
result match
    case Result.Error(ex: Throwable) => throw ex
    case Result.Error(_: ) => exit(1)
    case _               => ()
```

Reply:
This could work well :+1: 

I just want to point out that having to branch based on whether the error is `<: Throwable` is the unfortunate downside of allowing Result to have error which aren't `<: Throwable`. If all errors in were `<: Throwable`, we could `throw` all of them.